client/src/tutorials/TinyAdventureTwo/files/lib.rs

```diff
@@ -0,0 +1,143 @@
+            },
+        );
+        system_program::transfer(cpi_context, CHEST_REWARD)?;
+
+        msg!("Level Reset and Chest Spawned at position 3");
+
+        Ok(())
+    }
+
+    pub fn move_right(ctx: Context<MoveRight>) -> Result<()> {
+        let game_data_account = &mut ctx.accounts.game_data_account;
+        if game_data_account.player_position == 3 {
+            msg!("You have reached the end! Super!");
+        } else if game_data_account.player_position == 2 {
+            game_data_account.player_position = game_data_account.player_position + 1;
+
+            msg!(
+                "You made it! Here is your reward {0} lamports",
+                CHEST_REWARD
+            );
+
+            **ctx
+                .accounts
+                .chest_vault
+                .to_account_info()
+                .try_borrow_mut_lamports()? -= CHEST_REWARD;
+            **ctx
+                .accounts
+                .player
+                .to_account_info()
+                .try_borrow_mut_lamports()? += CHEST_REWARD;
+        } else {
+            game_data_account.player_position = game_data_account.player_position + 1;
+            print_player(game_data_account.player_position);
+        }
+        Ok(())
+    }
+}
+
+fn print_player(player_position: u8) {
+    if player_position == 0 {
+        msg!("A Journey Begins!");
+        msg!("o.........💎");
+    } else if player_position == 1 {
+        msg!("..o.......💎");
+    } else if player_position == 2 {
+        msg!("....o.....💎");
+    } else if player_position == 3 {
+        msg!("........\\o/💎");
+        msg!("..........\\o/");
+        msg!("You have reached the end! Super!");
+    }
+}
+
+#[derive(Accounts)]
+pub struct InitializeLevelOne<'info> {
+    // We must specify the space in order to initialize an account.
+    // First 8 bytes are default account discriminator,
+    // next 1 byte come from NewAccount.data being type u8.
+    // (u8 = 8 bits unsigned integer = 8 bytes)
+    // You can also use the signer as seed [signer.key().as_ref()],
+    #[account(
+        init_if_needed,
+        seeds = [b"level1"],
+        bump,
+        payer = signer,
+        space = 8 + 1
+    )]
+    pub new_game_data_account: Account<'info, GameDataAccount>,
+    // This is the PDA in which we will deposit the reward SOl and
+    // from where we send it back to the first player reaching the chest.
+    #[account(
+        init_if_needed,
+        seeds = [b"chestVault"],
+        bump,
+        payer = signer,
+        space = 8
+    )]
+    pub chest_vault: Account<'info, ChestVaultAccount>,
+    #[account(mut)]
+    pub signer: Signer<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct MoveRight<'info> {
+    #[account(mut)]
+    pub chest_vault: AccountInfo<'info>,
+    #[account(mut)]
+    pub game_data_account: Account<'info, GameDataAccount>,
+    #[account(mut)]
+    pub player: AccountInfo<'info>,
```

Comment:
`player` is a `Signer`

```rs
pub player: Signer<'info>,
```

Reply:
Hey I added the signer check to the account macro because the SystemProgram:Transfer expects an AccountInfo. I hope thats also ok.